### ISSUE TITLE
Very large history - GIT problem

### ISSUE THREAD

**Issue by pdoteter** [robiningelbrecht/strava-activities-template#169]
I have a very big profile (activities from 2007).  Did several runs, the last run finally got, 9 runs all hitting the API limits :)
https://www.strava.com/athletes/591020

The last one finally ran for the last activities, but the code goes through all activity streams
Logging like this:
2023-12-18T09:09:47.4960644Z   => Imported activity stream "1417229514 - heartrate"
I hit the data limit that day after 2 hours and with the push back to git it failed:

2023-12-18T09:10:08.5042734Z  1 file changed, 0 insertions(+), 0 deletions(-)
2023-12-18T09:10:22.6342486Z remote: error: Trace: 342832c495391c7cf2c6cfbf0303fd3c364dd005cc83d7bf7f19d042052255c7        
2023-12-18T09:10:22.6344516Z remote: error: See https://gh.io/lfs for more information.        
2023-12-18T09:10:22.6346333Z remote: error: File database/db.strava is 190.20 MB; this exceeds GitHub's file size limit of 100.00 MB        
2023-12-18T09:10:22.6349135Z remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.        
2023-12-18T09:10:22.6356141Z To https://github.com/pdoteter/Strava-activity
2023-12-18T09:10:22.6357745Z  ! [remote rejected] master -> master (pre-receive hook declined)
2023-12-18T09:10:22.6360113Z error: failed to push some refs to 'https://github.com/pdoteter/Strava-activity'

So it did a lot of work, but the DB is too large, don't even know if you there is a limit in storage space in github itself.

Going to look to try to run everything locally, I have an example of how everything gets scheduled and run.  It's a challenge ;)

**Comment by robiningelbrecht** [comment]
```
file database/db.strava is 190.20 MB; this exceeds GitHub's file size limit of 100.00 MB
```

So the reason is indeed you have too much activities and the size of the database exceeds GitHub's file size limit.
I'll run into this limit eventually as well I think. I could split op the database into several files to try to workaround this limit

**Comment by pdoteter** [comment]
I'm not very used to php and the used database, if it's possible to create the table linked to a file than at least you split the load across several files.  I think Microsoft SQL server can do that, not sure what the project is using.
Otherwise maybe 1 table = 1 database.
Is there a possibility to compress or clean the database, maybe some temporary space gets stored which can be purged before a run (transaction logs).  Maybe you can turn that off too.
I'm a business developer in .Net, so space is something I don't think about 😄

**Comment by robiningelbrecht** [comment]
@pdoteter I'm releasing an update that reduced the size of my personal db from 50mb to 20mb. This is of course a temporary fix. As time will go on, I'll run into the same issue as you did. So I need to think of a permanent fix to make this work with infinite amounts of data :)

This temp fix might reduce you DB by half as well and you'll be under 100mb...

**Comment by robiningelbrecht** [comment]
Reduced it further to 15MB. 
Got another improvement in mind that will reduce it to 13MB

**Comment by pdoteter** [comment]
I'll run the update again today, I did some reading too, found the attach possibility of sqllite too, but didn't found a direct way to use it with the Doctrine library

**Comment by robiningelbrecht** [comment]
@pdoteter What I'll probably do is split the "write" and "read" into two separate things.
Where the "write" part uses a database per year (all data for a particular year will end up in a separate DB). This will allow us to keep database files small and keep pushing them to GIT.
When running the script to generate the statistics, I'll just dump everything into one big database so I don't need to change any existing functionality BUT I will add this DB to `.gitignore` as we do not need keep track of it because we have the small "DBs per year"

What do you think?

**Comment by pdoteter** [comment]
@robiningelbrecht Good idea the per year database, the historical database won't change much too, it's a typical scenario which we also do in our business (data since 2004 which is just kept in offline readonline storage) but reading/writing is transparent for us, but the suggested solution to merge everything in a .gitignored database for the reading is a good way to not impact the reading part of the code.
BTW. My database now is 60Mb, stumbled upon the strava api limit again today (first run got an error on the strava gateway after 500 calls 😢 , going to activate the schedule again hopefully tomorrow a complete DB.

**Comment by robiningelbrecht** [comment]
@pdoteter ok, that's a reduction of 75% :). Good temporary fix. 
I'll get to the "DB per year" solution soon. For now, you'll be good with your 60mb DB :)
